### DISCUSSION TITLE
desktop file incorrect

### DISCUSSION THREAD

**Discussion by leoVici9** [sxyazi/yazi#2048] [1. Q&A]
### What system are you running Yazi on?

Linux Wayland

### What terminal are you running Yazi in?

foot

### `yazi --debug` output

```Shell
not relevant i Think
```

### Describe the question

Installation provides a deskop file ( on arch located at /usr/share/applications). This will make a yazi entry visible in application launchers  however yazi will not start due to incorrect exec line:
Exec=yazi %u

In order to succesfully start , the exec line needs to be changed to (foot specific)
Exec=foot -a yazi yazi %u

Not sure whether the terminal keyword in the desktop.ini should now be set to false iso true.

Question : Is the delivery of the desktop file intentional as it is ? or should there a a doc specifying the needed updates ?

### Checklist

- [X] I have searched the existing discussions/issues

**Comment by sxyazi** [comment]
I'm not sure if setting `Exec` to a specific terminal is the right way, I've never actually used that desktop file myself.

Maybe @fufexan can give some pointers, since they added it.

**Reply by fufexan** [reply]
@SolitudeSF's answer is correct. The desktop file is in accordance with the [Desktop Entry specification](https://specifications.freedesktop.org/desktop-entry-spec/latest/index.html).

**Reply by leoVici9** [reply]
Well that might be but the effect still is that it will not work consistently due to the fact that this [https://lists.freedesktop.org/archives/xdg/2020-July/014340.html](url) is only in proposal state for several years now and each de currently handles it differently. So nothing wrong with the launcher I am using but the assumption that such a correct desktop file will work is wrong.

**Reply by SolitudeSF** [reply]
If the correct desktop file doesn't work with your launcher, then the launcher is wrong. What even is the solution you're proposing?

**Reply by leoVici9** [reply]
I was trying to say that the desktop file spec might be correct but the implementation by various lauchers is not specified  (because still in proposal) , how can a launcher be wrong when there is no spec ?  For example gtk-launch starts yazi in gnome console, pop-launcher does not start it , walker and sirula start it in kitty ( thats my current terminal of use).  If theming is important to that user results are unexpected.

I started this as a discussion not as an issue , so i dont know the proper solution. It might be better to not supply the desktop file and just to document  

or  

to document that in certain cases a local desktop file with terminal is false and an exec string with kitty yazi  or whatever terminal you are using is a better solution.

**Reply by fufexan** [reply]
> It might be better to not supply the desktop file and just to document

I'm not sure removing a feature people are already accustomed to is a good idea. I've been using it ever since I added it. It's how I designate yazi as the default file manager, which any app will open in a unified way (XDG associations).

Documentation would be nice to have, sure.

**Reply by sxyazi** [reply]
Doc sounds good to me, we can put it on the [FAQ page](https://yazi-rs.github.io/docs/faq) I guess, @bertSN1601 would you like to raise a PR for that?

**Comment by SolitudeSF** [comment]
Desktop file is correct. Your application launcher is misconfigured. Setting `Terminal=true` is a proper way to make an app launch in the terminal.

**Reply by SolitudeSF** [reply]
Try setting `TERMINAL` env variable in your `.profile` to your preferred terminal.

**Comment by sxyazi** [comment]
Gotcha, thanks for confirming! @SolitudeSF @fufexan
